azure-pipelines/pre-release-pipelines.yml

```diff
@@ -0,0 +1,78 @@
+trigger:
+- master
+
+variables:
+  NodeVersion: '10.x'
+
+jobs:
+  - job: 'Build'
+
+    pool:
+      vmImage: 'Ubuntu 16.04'
+
+    steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: '$(NodeVersion)'
+        displayName: 'Install Node.js $(NodeVersion)'
+
+      - task: Npm@1
+        displayName: 'npm install'
+        inputs:
+          verbose: false
+
+      - script: 'gulp build'
+        displayName: 'gulp build'
+
+      - script: 'gulp test'
+        displayName: 'gulp test'
+        
+      - script: 'gulp dotnet/pack'
+        displayName: 'gulp dotnet/pack'
+        
+      - task: EsrpCodeSigning@1
+        inputs:
+          ConnectedServiceName: $(esrp-connection)
+          FolderPath: 'src/dotnet/AutoRest/bin/netcoreapp2.0'
+          Pattern: '*.dll,*.exe'
+          signConfigType: 'inlineSignParams'
+          inlineOperation: |
+            $(inline-operation-param)
+          SessionTimeout: '60'
+          MaxConcurrency: '50'
+          MaxRetryAttempts: '5'
+
+      - script: 'gulp pack'
+        displayName: 'gulp pack'
+
+      - task: PublishBuildArtifacts@1
+        inputs:
+          pathtoPublish: $(Build.SourcesDirectory)/dist
+
+  - job: Windows
+    displayName: credentialScan
+    pool:
+      vmImage: "windows-2019"
+
+    steps:
+      - task: ea576cd4-c61f-48f8-97e7-a3cb07b90a6f@2 # CredScan@2  
```

Comment:
Could you use a user friendly name for this task?

Reply:
there is a issue in DevOps  cause the task name is always ambiguous , so here,  provided a guid instead of task name to avoid it